src/components/interaction-designer/block-editors/choices/mixins/CommonVoiceChoiceConfig.vue

```diff
@@ -0,0 +1,81 @@
+<script>
+import {mapActions} from 'vuex'
+import {IChoice, ISelectOneResponseBlock} from '@floip/flow-runner'
+import {BLOCK_TYPE} from '@/store/flow/block-types/MobilePrimitives_SelectOneResponseBlockStore'
+
+export default {
+  props: {
+    block: {
+      type: ISelectOneResponseBlock,
+      required: true,
+    },
+    choice: {
+      type: IChoice,
+      required: true,
+    },
+    index: {
+      type: Number,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      options: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0', '*'],
+      blockResponseExpression: 'block.response',
+    }
+  },
+  computed: {
+    currentExpressionWithoutSpaces() {
+      return this.currentExpression !== undefined ? this.currentExpression.replace(/\s/g, '') : undefined
+    },
+    currentExpression() {
+      return this.choice.ivr_test?.test_expression
+    },
+    /**
+     * Test if:
+     * - the expression starts with `block.response='` (floip schema) or `@block.response='` (from partner's flow)
+     * - and if the estimated key_press is part of the options' elements
+     */
+    isKeyPressIdentifiable() {
+      if (this.currentExpressionWithoutSpaces?.startsWith(`${this.blockResponseExpression}='`) === true
+        || this.currentExpressionWithoutSpaces?.startsWith(`@${this.blockResponseExpression}='`) === true) {
+        const estimatedKeyPress = this.currentExpressionWithoutSpaces.split('\'')?.[1]
+        if (this.options.includes(String(estimatedKeyPress)) === true) {
+          return true
+        }
+      }
+
+      return false
+    },
+    estimatedKeyPress() {
+      return this.currentExpressionWithoutSpaces?.split('\'')?.[1]
+    },
+    keypress: {
+      get() {
+        if (this.isKeyPressIdentifiable === true) {
+          return String(this.estimatedKeyPress)
+        }
+
+        // Provide an incremental value based on existing choices
+        // For now, just a simple incremental logic without considering the choice's ivr_test value
```

Comment:
- [ ] Apart from this I will make sure to uncheck `use expression` checkbox by default

Reply:
Probably by setting `ivr_test.test_expression = ''` when loading the component (if IVR mode is enabled)